User authentication to perform actions on articles and comments

miguel0577 opened the pull request:
## What?
Added a user authentication feature for creating and editing articles and comments in the blog project.

## Why?
An unregistered user cannot make changes to the blog,  so a feature was implemented to limit these actions only to authorized users.

## How?
Implemented the Devise gem in Rails for authentication and added a user model for user identification and control

## Tests?
The operation of the application with the remote server was verified after a Deploy to Heroku

## Screenshots
![image](https://user-images.githubusercontent.com/108099572/194691327-591dfc3a-8285-47a8-83e3-7b3f33fb3bb6.png)
![image](https://user-images.githubusercontent.com/108099572/194691354-332c525f-2eef-469e-b830-c5ac04a00690.png)

## Anything else?
Soon the function to follow and unfollow other bloggers will be implemented!

miguel0577 committed:
Add comments only for logged users

miguel0577 committed:
Show bloggers only if the user is logged in

estebancordoba commented on app/views/articles/show.html.erb:
```diff
@@ -1,17 +1,32 @@
-<h2>Comments</h2>
-<%= render @article.comments %>
+    <% if user_signed_in?  %>
+      <% if @article.owner == current_user.id %>
```
Could we move this condition to a method and then call the method on the view? Just like you do with `user_signed_in?`

miguel0577 replied:
Yes, this condition could be moved to a method. I will add this in the next pull request

estebancordoba replied:
The best thing would be to add that change in this PR, so I can check the changes, the same for the comment https://github.com/miguel0577/blog/pull/12#discussion_r991333350

estebancordoba commented on app/views/devise/mailer/reset_password_instructions.html.erb:
```diff
@@ -0,0 +1,8 @@
+<p>Hello <%= @resource.email %>!</p>
```
Is the feature to send mails functional?

miguel0577 replied:
This file was added automatically using the devise gem, at the moment it is not functional, I will remove this file from the project

estebancordoba commented on app/models/user.rb:
```diff
@@ -0,0 +1,6 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
```
What does each of these options mean?

miguel0577 replied:
They are the modules that can be configured from the Devise method in the User model in this case, they are:

database_authenticatable, with this configuration users, will be able to authenticate with a username and password stored in the database.

registerable, so that users can register, update and delete their profiles.

rememberable, to enable the option to remember me on the login

validatable, which validates the User's email and password

estebancordoba commented on app/views/articles/_form.html.erb:
```diff
@@ -32,14 +32,15 @@
         </div>
 
         <div class="label-submit-article">
+          <% form.select :owner, [current_user.id] %>
```
Why is it sent with a select? It should be sent with a hidden field

miguel0577 replied:
It´s true, sorry, it was a provisional option since it did not take into account all the options of the form, I will make this change for the next pull request, thanks for the advise

estebancordoba replied:
Like the previous comments, these types of changes are made in the same PR since they are part of the same feature

miguel0577 committed:
Recover account feature removed

The feature to recover accounts with mail delivery has been deleted

miguel0577 committed:
Changed form function (owner) 

The way of sending the owner data was changed, now this value is sent with a form.hidden_field

miguel0577 committed:
Non-functional files removed

The files that render the recovering user by mail function were removed from the project

miguel0577 committed:
Non-functional files deleted

The files that render the recovering user by mail function were removed from the project

miguel0577 committed:
Non-functional files deleted

The files that render the recovering user by mail function were removed from the project

miguel0577 committed:
Non-functional files deleted

The files that render the recovering user by mail function were removed from the project

miguel0577 committed:
Non-functional files deleted

The files that render the recovering user by mail function were removed from the project

miguel0577 committed:
Condition to add comments changed

This was the only way I was able to adapt the conditional outside of the views file, the condition was added to the articles controller file.

miguel0577 committed:
Condition to add comments changed (controller)

The conditional to determine if the current user is the owner of the article being viewed is added to this controller

miguel0577 committed:
Merge branch 'bootstrap-1' into show-logged-bloggers
